crates/common/src/provider/runtime_transport.rs

```diff
@@ -64,6 +73,7 @@ pub enum RuntimeTransportError {
 /// A runtime transport is a custom [alloy_transport::Transport] that only connects when the *first*
 /// request is made. When the first request is made, it will connect to the runtime using either an
 /// HTTP WebSocket, or IPC transport depending on the URL used.
+/// It also supports retries for rate-limiting and timeout-related errors.
 #[derive(Clone, Debug, Error)]
 pub struct RuntimeTransport {
```

Comment:
Discussed offline, this will eventually become a service reusable in alloy, but we need further changes in Alloy before we can move forward to making this a service